cmd/activator/main.go

```diff
@@ -273,16 +281,53 @@ func main() {
 			}
 		}(name, server)
 	}
-	// Exit as soon as we see a shutdown signal or one of the servers failed.
+
+	// Wait for the signal to drain.
 	select {
-	case <-ctx.Done():
+	case <-drainCh:
+		logger.Info("Received the drain signal.")
 	case err := <-errCh:
 		logger.Errorw("Failed to run HTTP server", zap.Error(err))
 	}
 
+	// The drain has started (we are now failing readiness probes).  Let the effects of this
+	// propagate so that new requests are no longer routed our way.
+	time.Sleep(30 * time.Second)
+	logger.Info("Done waiting, shutting down servers.")
+
+	// Drain outstanding requests, and stop accepting new ones.
 	for _, server := range servers {
 		server.Shutdown(context.Background())
 	}
+	logger.Info("Servers shutdown.")
+}
+
+func healthCheck(logger *zap.SugaredLogger, statSink *websocket.ManagedConnection) (func() error, <-chan struct{}) {
```

Comment:
The name of the function leads you to think that we will check health in this method, but we only set up healthCheck, so perhaps: `setupHealthCheck`?

Reply:
Made it `newHealthCheck`.